ironic/tests/unit/drivers/modules/drac/test_management.py

```diff
@@ -445,6 +445,25 @@ def test_set_boot_device_with_invalid_job_queue(
         self.assertEqual(0, mock_client.set_bios_settings.call_count)
         self.assertEqual(0, mock_client.commit_pending_bios_changes.call_count)
 
+    @mock.patch.object(drac_job, 'list_unfinished_jobs', spec_set=True,
+                       autospec=True)
+    def test_set_boot_device_with_unfinished_jobs(
+            self, mock_list_unfinished_jobs, mock_get_drac_client):
+        mock_client = mock.Mock()
+        mock_get_drac_client.return_value = mock_client
+
+        bios_job_dict = {
+            'id': 'JID_602553293345',
+            'name': 'ConfigBIOS:BIOS.Setup.1-1',
+            'start_time': 'TIME_NOW',
+            'until_time': 'TIME_NA',
+            'message': 'Task successfully scheduled.',
+            'status': 'Scheduled',
+            'percent_complete': 0}
+        bios_job = test_utils.dict_to_namedtuple(values=bios_job_dict)
+        mock_list_unfinished_jobs.return_value = bios_job
+        mock_client.delete_jobs(job_ids=['JID_CLEARALL_FORCE'])
```

Comment:
Instead of the above line, I think we want to call set_boot_device and then assert that delete_jobs was called.

Reply:
Calling delete_jobs on mock_client doesn't test any code because that is a mocked method.  We should call set_boot_device instead, then assert that delete_jobs was called with a parameter of ['JID_CLEARALL_FORCE'].  We should also have pretty much the same asserts that are in the test_set_boot_device test case.

Reply:
done